docs/html/reference/pip_show.rst

```diff
@@ -25,10 +25,49 @@ Options
 .. pip-command-options:: show
 
 
+Format
+======
+
+The various fields present in the JSON output and their explanation is as follows.
+
+*metadata*
+   A dictionary with the core metadata fields present in the ``METADATA`` file,
+   as defined in the `Core metadata specifications`_. We use the method detailed in
+   `JSON-compatible Metadata`_ to convert core metadata to json. The fields are
+   lower cased, with dashes replaced by underscores.
+
+*direct_url*
+   A dictionary containing the content of ``direct_url.json``,
+   if present, as specified in :pep:`610`.
+
+*installer*
+   A string containing the content of ``INSTALLER``,
+   if present, as specified in :pep:`376`.
+
+*record*
+   A list of ``[file_path, file_content_hash, file_size]``, representing
+   the content of ``RECORD``, if present, as specified in :pep:`376`.
+
+*requested*
+   A boolean, set to True if ``REQUESTED`` is present,
+   as specified in :pep:`376`.
+
+*required_by*
+   A list of canonicalized distribution names that depend
+   on the queried distribution.
+
+*location*
+   A string containing the path where the distribution is installed. For legacy installs,
+   this is the parent directory of the metadata (.dist-info or .egg-info) directory.
+   For legacy editable installs, this is the directory where the source is located.
```

Comment:
This could be simplified by saying that location is the parent of the metadata (dist-info or egg-info) directory. This holds True for all kinds of installation.

Reply:
Added in https://github.com/pypa/pip/pull/8008/commits/02759a3e87811248b6a9bc4e5db4ccdbdc6bf1b5